File: api/src/Domain/Billing/ValueObjects/SubscriptionPlanPrice.cs
using System;

namespace DetailingArsenal.Domain.Billing {
    public class SubscriptionPlanPrice : ValueObject<SubscriptionPlanPrice> {
        public decimal Price { get; }
        public string Interval { get; }
        public BillingReference BillingReference { get; }

        public SubscriptionPlanPrice(decimal price, string interval, BillingReference billingReference) {
            Price = price;
            Interval = interval;
            BillingReference = billingReference;
        }
    }
}

File: app/src/api/billing/data-transfer-objects/subscription-plan-price.ts
import { BillingInterval } from '@/api/billing/data-transfer-objects/billing-interval';

/**
 * A price for a subscription plan.
 */
export class SubscriptionPlanPrice {
    constructor(public amount: number = 0, public interval: BillingInterval, public billingId: string) {}
}
